src/Featurizers/NormalizeFeaturizer.h

```diff
@@ -0,0 +1,373 @@
+    typename IteratorRangeT,
+    typename UpdaterType,
+    size_t MaxNumTrainingItemsV=std::numeric_limits<size_t>::max()
+>
+class NormalizeEstimatorImpl : public TransformerEstimator<IteratorRangeT, std::vector<std::double_t>> {
+public:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Types
+    // |
+    // ----------------------------------------------------------------------
+    using TransformedType                   = std::vector<std::double_t>;
+    using BaseType                          = TransformerEstimator<IteratorRangeT, TransformedType>;
+    using TransformerType                   = NormalizeTransformer<IteratorRangeT>;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Public Methods
+    // |
+    // ----------------------------------------------------------------------
+    NormalizeEstimatorImpl(AnnotationMapsPtr pAllColumnAnnotations, size_t colIndex);
+    ~NormalizeEstimatorImpl(void) override = default;
+
+    FEATURIZER_MOVE_CONSTRUCTOR_ONLY(NormalizeEstimatorImpl);
+
+private:
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Data
+    // |
+    // ----------------------------------------------------------------------
+    size_t const                            _colIndex;
+
+    // ----------------------------------------------------------------------
+    // |
+    // |  Private Methods
+    // |
+    // ----------------------------------------------------------------------
+    bool begin_training_impl(void) override;
+    FitResult fit_impl(IteratorRangeT const *, size_t) override;
+    void complete_training_impl(void) override;
+
+    // MSVC has problems when the definition is separate from the declaration
+    typename BaseType::TransformerUniquePtr create_transformer_impl(void) override {
+        // ----------------------------------------------------------------------
+        using VectorNormsAnnotationData = Components::VectorNormsAnnotationData<UpdaterType>;
+        using VectorNormsEstimator       = Components::VectorNormsEstimator<IteratorRangeT, UpdaterType, MaxNumTrainingItemsV>;
+        // ----------------------------------------------------------------------
+
+        VectorNormsAnnotationData const & data(VectorNormsEstimator::get_annotation_data(this->get_column_annotations(), _colIndex, Components::VectorNormsEstimatorName));
+
+        return typename BaseType::TransformerUniquePtr(new NormalizeTransformer<IteratorRangeT>(data.Norms));
+    }
+};
+
+} // namespace Details
+
+/////////////////////////////////////////////////////////////////////////
+///  \class         NormalizeEstimator
+///  \brief         Chains up VectorNormsEstimator and NormalizeTransformerImpl
+///                 to normalize input vector data so that it's norm becoming 1
+///
+template <
+    typename IteratorRangeT,
+    typename UpdaterType,

```

Comment:
>typename UpdaterType, [](start = 4, length = 21)

The code gen code won't be able to instantiate this Estimator, as it won't be able to determine what this value should be. If you need to distinguish between L1 and L2 functionality, you might be looking at 2 different Estimators.